driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1362,16 +1374,17 @@ synchronized void init() {
 
             this.configuration.register(this);
 
-            this.executorQueue = new LinkedBlockingQueue<Runnable>();
-            this.executor = makeExecutor(NON_BLOCKING_EXECUTOR_SIZE, "worker", executorQueue);
-            this.blockingExecutorQueue = new LinkedBlockingQueue<Runnable>();
-            this.blockingExecutor = makeExecutor(2, "blocking-task-worker", blockingExecutorQueue);
-            this.reconnectionExecutor = new ScheduledThreadPoolExecutor(2, threadFactory("reconnection"));
-            // scheduledTasksExecutor is used to process C* notifications. So having it mono-threaded ensures notifications are
-            // applied in the order received.
-            this.scheduledTasksExecutor = new ScheduledThreadPoolExecutor(1, threadFactory("scheduled-task-worker"));
-
-            this.reaper = new ConnectionReaper(this);
+            ThreadingOptions threadingOptions = this.configuration.getThreadingOptions();
+            ThreadPoolExecutor tmpExecutor = threadingOptions.executor(clusterName);
+            this.executorQueue = tmpExecutor.getQueue();
+            this.executor = MoreExecutors.listeningDecorator(tmpExecutor);
```

Comment:
I think the need for monitoring thread pool queues is even greater now that the users have control over configuring them, so I think it would be good to keep it (the cost is low anyways).   Also think we should not remove the metrics in a patch release as it could break some users existing monitoring (i.e. if they were alerting on these metrics, although that likelihood seems low).